Add a contributor path (#124)
cncf/tag-env-sustainability: governance/contributor-ladder.md

> # Outlining your Contributor Path

Reviewer: suggestion:  "Contributor Path" 

Simpler?

  - [Leading a project](#leading-a-project)
  - [Helping out](#helping-out)
  - [Leading a Working Group](#leading-a-working-group)
  - [Nominated as Tech Lead](#nominated-as-tech-lead)
  - [Nominated as Co-Chair](#nominated-as-co-chair)

## Starting out

When venturing into the open source space, it's crucial to familiarize yourself with the way we operate and collaborate as a global community.
Take some time to reflect on your personal journey.
Find out if you genuinely derive joy from this line of work and consider what areas of knowledge you are particularly curious about.
Remember, actively participating and providing valuable input are significant aspects of your involvement in the community.
By engaging with others and sharing your insights, you contribute to the collective growth and success of the open source ecosystem.

* Join Community Meetings, the Slack Channel and the mailing list (take a look at [this page](https://github.com/cncf/tag-env-sustainability/#contact))
* Proactively participate in discussions at community meetings, in Slack threads, or [GitHub issues](https://github.com/cncf/tag-env-sustainability/issues).

> * Reflect on your personal journey spent at TAG

Reviewer: I'd remove the "reflect" one.  I'm not sure what it adds here.

Author: Since this is the first step in the journey, people don't know if they like the subject, the group, the foundation, the way we work, and so on. This item mentions that you should think about whether the way we work suits you. 

I will remove it as it is not a clear action. And ... You will, of course, move away from the group if it does not suit you ....

* Get to know the Group

## Taking ownership & responsibility

> Embracing more leadership within the Environmental Sustainability TAG is highly encouraged. Actively participating in discussions and driving ideas and projects forward is crucial for the group's advancement.

Reviewer: ```suggestion
Providing leadership within the Environmental Sustainability TAG is highly encouraged. Actively participating in discussions and driving ideas and projects forward is crucial for the group's advancement.
```

Within the [`roles.md`](roles.md) file, you will find various volunteer positions available.
To begin, consider taking ownership of a small initiative within the TAG, such as opening an issue that was discussed during a meeting, or taking on responsibilities like reviewing pull requests (PRs).
Additionally, you can contribute by raising awareness about the TAG ENV among your network. This can involve sharing information and updates, presenting talks at conferences or events, organizing meetups, or even writing a blog post (refer to [blog post process](blogpost-process.md) for guidance).
Collaboration with other organizations is also a priority for us, so actively seeking out open source projects or organizations you know in the cloud native sustainability space and connecting them with the TAG would be highly valuable.
By proactively driving topics that directly benefit the Environmental Sustainability TAG, you significantly contribute to its growth and overall impact and will help you mature as a contributor.

* Proactively start discussions and take ownership of smaller initiatives
* Establish yourself in the community

> * Get an overview of processes

Reviewer: I'd probably remove this line and the one beforehand and split up the other.  So maybe change this list to:

* Proactively start discussions.
* Take ownership of small initiatives.
* Provide help when help is needed or requested (for instance, taking notes or reviewing documents)
* Provide productive feedback to additions or subtractions to make processes more efficient and less work for involved parties.

Author: 👍 

changed the last one to `* Provide actionable feedback to processes.`

## Leading a project

Once you have established connections within the TAG and actively participated in discussions and projects, you may encounter a topic that captures your interest and warrants further attention.
This particular project might require more time and dedication from your side as well as the support of other individuals within the community.
If you have a project idea that aligns with the objectives of the TAG, consider writing a proposal, in a GitHub issue, outlining its motivation, scope, goals, and deliverables.
Present this proposal during a TAG meeting to initiate a discussion and gather feedback and support from fellow members.
Taking on a [project leadership role](roles.md#role-of-project-leads) within the community project becomes your next significant step towards leadership recognition within the group.

* Take ownership of a project within the TAG
* Lead a community group within the TAG

## Helping out

> In open source, newcomers join the community seeking guidance and support as they embark on their journey, just as you once did.

Reviewer: ```suggestion
In open source, newcomers join the community need guidance and support as they embark on their journey, just as you once did.
```

> While it is essential to provide support to current members and leaders, it is equally important to focus on assisting and orienting newcomers.

Reviewer: ```suggestion
While it is essential to provide support to current members and leaders, it is also important to focus on assisting and orienting newcomers.
```

Modeling empathy and offering support to the community as a whole should be at the forefront of your actions.

> It is crucial to embody the values of the community as outlined in the [code of conduct](http://cncf.io/conduct/).

Reviewer: ```suggestion
It is crucial to role model the values of the community as outlined in the [code of conduct](http://cncf.io/conduct/).
```

There are various ways you can contribute to supporting and empowering new contributors.

> Firstly, you can actively participate by stepping in and offering mentorship and guidance when someone requires assistance. This could involve helping them navigate the community, providing technical guidance, or answering their questions.

Reviewer: ```suggestion
You can actively participate by offering mentorship and guidance when someone requests assistance. This could involve helping them navigate the community, providing technical guidance, or answering their questions.
```

Additionally, if a member is unable to fulfill their responsibilities, you can offer to fill in their role temporarily or assist in any necessary tasks.
Co-hosting meetups and events is another way to provide support. By sharing the responsibility, you help create an inclusive environment where everyone feels supported and encouraged.

> Lastly, building a habit of consistently helping out within the TAG is highly valuable.

Reviewer: ```suggestion
Building a brand of being someone who consistently helps is highly valuable.
```

> This can include engaging in discussions, providing feedback on proposals, reviewing code contributions, or offering assistance in any way that aligns with your skills and expertise.

Reviewer: ```suggestion
Actions to help this include engaging in discussions, providing feedback on proposals, reviewing code contributions, and offering assistance in any way that aligns with your skills and expertise.
```

By actively participating and contributing, you demonstrate your commitment to the community and showcase the values that the Environmental Sustainability TAG holds dear.

* Helping out and mentor new contributors in their open source journey.
* Helping out community members and leaders if they need support.
* Demonstrating empathy and embody the community values.

## Leading a Working Group

Within the Environmental Sustainability TAG, various ongoing initiatives are carried out through dedicated working groups.
As leadership roles naturally transition over time, it is worthwhile for you to consider taking up a leadership position within an established working group.

> Alternatively, you can contribute by proposing the formation of a new working group.

Reviewer: ```suggestion
However, if the current working groups does not include scope for the item you are interested in, you could form a new working group.
```

> To do so, create a GitHub issue where you outline the motivation, scope, goals, and deliverables of the proposed working group.

Reviewer: I'd probably swap this bullet and the one after.  I usually have discussion to see if I have sufficient backing before making a new group/issue/initiative.

Author: Changed the section to this.. 😅 

> Since leadership roles naturally change over time, it is worth your while to take a leadership position in an established working group. Within the TAG for Environmental Sustainability, dedicated working groups carry out several ongoing initiatives. However, if the current working groups do not provide space for the topic you are interested in, you could start a new working group. A working group must have community support and requires that you find sufficient backing before a working group is formed. To start this process, create a GitHub issue describing the initial charter in which you outline the motivation, scope, goals, and deliverables of the proposed working group. This working group charter document should form the basis for discussions with the TAG before initiating an official vote by the TAG co-chairs, which also requires approval by the CNCF TOC. It is important to present the working group proposal during TAG meetings and be vocal about the discussions, actively solicit feedback and input, and gain support from other members.

Present this proposal during a TAG meeting to initiate a discussion, gather feedback, and garner support from fellow members.

* Taking on more responsibility and ownership within TAG.
* Leading a community group within TAG

> * Growing as a community leader

Reviewer: I'd probably move the "growing as a community leader" from here.  That's hard to quantify.

Author: 👍

Your position as a technical lead allows you to contribute significantly to the overall success and cohesion of the Environmental Sustainability TAG.

* Build partnerships between TAG and projects or other organizations.
* Support and promote other community leaders

## Nominated as Co-Chair

To serve in the role of Co-Chair, you have to go through the [chair proposal process](https://github.com/cncf/tag-env-sustainability/blob/main/governance/chair-proposal-process.md).
With your appointment as Co-Chair of TAG for Environmental Sustainability, you assume a key leadership position within the Group.
Your will steer and evolve the community to achieve its goals and targets. 
You coordinate between the TAG and the CNCF Technical Oversight Committee (TOC) and the overall CNCF ecosystem, ensuring effective communication and alignment of efforts.
You will support and uplift other community leaders and provide guidance, mentorship, and opportunities for growth, you help cultivate a vibrant and empowered leadership team within the TAG.

* Driving the continuous evolving of the TAG
* Steering the TAG
* Coordinating between the TAG and the TOC
* Support and uplift other community leaders

> * Having joy seeing others progressing as leaders in the TAG

Reviewer: I'd remove this last one.  It's not quantifiable.  

Maybe add, however, collaborating and steering new members into leading new positions on the tag, or something similar to that.

## Leading a Working Group

Since leadership roles naturally change over time, it is worth your while to take a leadership position in an established working group.
Within the TAG for Environmental Sustainability, dedicated working groups carry out several ongoing initiatives.
However, if the current working groups do not provide space for the topic you are interested in, you could start a new working group.
A working group must have community support and requires that you find sufficient backing before a working group is formed.
To start this process, create a GitHub issue describing the initial charter in which you outline the motivation, scope, goals, and deliverables of the proposed working group.
This working group charter document should form the basis for discussions with the TAG before initiating an official vote by the TAG co-chairs, which also requires approval by the CNCF TOC.
It is important to present the working group proposal during TAG meetings and be vocal about the discussions, actively solicit feedback and input, and gain support from other members.

* Taking on more responsibility and ownership within TAG.
* Leading a community group within TAG.

> * Conflict resolution.

Reviewer: These three bullets kinda are sitting there without a lot of context, now that I'm re-reading.  Do we want to say "pre-requisites for leaders"?

```suggestion
* Good at negotiating with differing opinions.
```

I try not to suggest there will be conflicts, but it is good to have something similar to that here.

## Nominated as Tech Lead

To serve in the role as Tech Lead, you have to go through the [tech lead proposal process](https://github.com/cncf/tag-env-sustainability/blob/main/governance/tech-lead-proposal-process.md).
Assuming the role of a technical lead within the Environmental Sustainability TAG places greater responsibilities on your shoulders as part of the leadership team.
This roles provides you with the opportunity to actively support the leadership team by taking charge of a specific topic within the TAG.
As an elected Tech Lead and a member of the leadership team, you will have the chance to delve into the broader aspects of leading the community group, gaining insights into the organizational structures within the CNCF. 
Additionally, you will play a crucial role in facilitating discussions and resolving any issues that may arise between contributors, ensuring a harmonious and productive community environment.
Your position as a technical lead allows you to contribute significantly to the overall success and cohesion of the Environmental Sustainability TAG.

> * Build partnerships between TAG and projects or other organizations.

Reviewer: This is "expectations of a tech lead?"

Author: Removed it, @catblade do you may have a suggestion what we could add here? We just have one bullet point.

Reviewer: Good point.  Can we move some of the content from the above paragraphs into the bullet points?

Like:
* Facilitating discussions
* Chop wood/carry water (or whatever phrase that Max likes)
* Lead WGs and make sure deliverables stay on schedule

* Support and promote other community leaders.

## Nominated as Co-Chair

To serve in the role of Co-Chair, you have to go through the [chair proposal process](https://github.com/cncf/tag-env-sustainability/blob/main/governance/chair-proposal-process.md).
With your appointment as Co-Chair of TAG for Environmental Sustainability, you assume a key leadership position within the Group.
Your will steer and evolve the community to achieve its goals and targets. 
You coordinate between the TAG and the CNCF Technical Oversight Committee (TOC) and the overall CNCF ecosystem, ensuring effective communication and alignment of efforts.
You will support and uplift other community leaders and provide guidance, mentorship, and opportunities for growth, you help cultivate a vibrant and empowered leadership team within the TAG.

> * Driving the continuous evolving of the TAG.

Author: What would you like to add specifically? A list header?

> * Driving the continuous evolving of the TAG.

Reviewer: Yep.  Something to make it easy for people to see requirements.  Do you have a better way of doing it (happy to see other ways).

> * Driving the continuous evolving of the TAG.

Author: added

## Nominated as Tech Lead

To serve in the role as Tech Lead, you have to go through the [tech lead proposal process](https://github.com/cncf/tag-env-sustainability/blob/main/governance/tech-lead-proposal-process.md).
Assuming the role of a technical lead within the Environmental Sustainability TAG places greater responsibilities on your shoulders as part of the leadership team.
This roles provides you with the opportunity to actively support the leadership team by taking charge of a specific topic within the TAG.
As an elected Tech Lead and a member of the leadership team, you will have the chance to delve into the broader aspects of leading the community group, gaining insights into the organizational structures within the CNCF. 
Additionally, you will play a crucial role in facilitating discussions and resolving any issues that may arise between contributors, ensuring a harmonious and productive community environment.
Your position as a technical lead allows you to contribute significantly to the overall success and cohesion of the Environmental Sustainability TAG.

* Support and promote other community leaders.
* Facilitating discussions

> * Chop wood/carry water (or whatever phrase that Max likes)

Reviewer: ```suggestion
* Carry water/gather food
```

# Outlining your Contributor Path

> The Environmental Sustainability TAG aims to support individuals who are passionate about cloud native sustainability and provide a space for personal growth as contributors. Within the [`roles.md`](roles.md) file, you will find various volunteer positions available. This document outlines a path for advancing as a contributor in the TAG. It's important to note that your journey may not precisely align with this path, as the document serves as a general guide to help you progress gradually. If you have already made significant contributions to another TAG or a project within the CNCF landscape and are familiar with the community, your path will naturally differ from that of someone who is just starting out.

Reviewer: Nit, but can we break this line up into multiple lines?  Ease for when we need to change something later.

Adding in some wording suggestions here:

The CNCF Environmental Sustainability TAG aims to support individuals who are passionate about cloud native sustainability and provide a space for personal growth as contributors. This document outlines a path for advancing as a contributor in the TAG. It's important to note that your journey may not precisely align with this path, as the document serves as a general guide. If you have already made significant contributions to another TAG or a project within the CNCF landscape and are familiar with the community, your path will naturally differ from that of someone who is just starting out.

* Join Community Meetings, the Slack Channel and the mailing list (take a look at [this page](https://github.com/cncf/tag-env-sustainability/#contact))
* Proactively participate in discussions at community meetings, in Slack threads, or [GitHub issues](https://github.com/cncf/tag-env-sustainability/issues).
* Reflect on your personal journey spent at TAG
* Get to know the Group

## Taking ownership & responsibility

Embracing more leadership within the Environmental Sustainability TAG is highly encouraged. Actively participating in discussions and driving ideas and projects forward is crucial for the group's advancement.

> To begin, consider taking ownership of a small initiative within the TAG, such as opening an issue that was discussed during a meeting, or taking on responsibilities like reviewing pull requests (PRs). Additionally, you can contribute by raising awareness about the TAG ENV among your network. This can involve sharing information and updates, presenting talks at conferences or events, organizing meetups, or even writing a blog post (refer to [blog post process](blogpost-process.md) for guidance). Collaboration with other organizations is also a priority for us, so actively seeking out open source projects or organizations you know in the cloud native sustainability space and connecting them with the TAG would be highly valuable.

Reviewer: I'd move the mention of roles to down here.

Author: 👍

By proactively driving topics that directly benefit the Environmental Sustainability TAG, you significantly contribute to its growth and overall impact and will help you mature as a contributor.

* Proactively start discussions and take ownership of smaller initiatives
* Establish yourself in the community
* Get an overview of processes

## Leading a project

> Once you have established connections within the TAG and actively participated in discussions and projects, you may encounter a topic that captures your interest and warrants further attention. This particular project might require more time and dedication from your side, as well as the support of other individuals within the community. If you have a project idea that aligns with the objectives of the TAG, consider writing a proposal, in a GitHub issue, outlining its motivation, scope, goals, and deliverables. Present this proposal during a TAG meeting to initiate a discussion and gather feedback and support from fellow members. Taking on a [project leadership role](roles.md#role-of-project-leads) within the community project becomes your next significant step towards personal growth within the group.

Reviewer: I would remove the word "personal growth" and replace with "leadership recognition" because it is more geared towards the group.  Personal growth has a much broader definition.

Reviewer: Add grammar also, remove comma from this sentence:  This particular project might require more time and dedication from your side as well as the support of other individuals within the community

Author: 👍

* Take ownership of a project within the TAG
* Lead a community group within the TAG

> * Grow as a community leader

Reviewer: This last one-is it supposed to be after one of the first two bullets?  Confused as to its purpose.

Author: Removed the bullet point

## Helping out

> In the open source field, newcomers continuously join the community seeking guidance and support as they embark on their journey, just as you did once. While it is essential to provide support to current members and leaders, it is equally important to focus on assisting and orienting new individuals. Demonstrating empathy and offering support to the community as a whole should be at the forefront of your actions. It is crucial to embody the values of the community as outlined in the [code of conduct](http://cncf.io/conduct/).

Reviewer: In open source, newcomers join the community seeking guidance and support as they embark on their journey, just as you once did. While it is essential to provide support to current members and leaders, it is equally important to focus on assisting and orienting newcomers. Modeling empathy and offering support to the community as a whole should be at the forefront of your actions. It is crucial to embody the values of the community as outlined in the [code of conduct](http://cncf.io/conduct/).

Author: 👍
